java-checks/src/test/java/org/sonar/java/checks/spring/StaticFieldInjectionNotSupportedCheckTest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonar.java.checks.spring;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.java.checks.verifier.CheckVerifier;
+
+import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+
+class StaticFieldInjectionNotSupportedCheckTest {
+
+  @Test
+  void test_compiling(){
+    CheckVerifier.newVerifier()
+      .onFile(mainCodeSourcesPath("checks/spring/StaticFieldInjectionNotSupportedCheckSample.java"))
+      .withCheck(new StaticFieldInjectionNotSupportedCheck())
+      .verifyIssues();
+  }
```

Comment:
I would add a unit test to guarantee that we do not raise issues when we don't know where annotations are coming from
```
  @Test
  void test_no_semantics(){
    CheckVerifier.newVerifier()
      .onFile(mainCodeSourcesPath("checks/spring/StaticFieldInjectionNotSupportedCheckSample.java"))
      .withCheck(new StaticFieldInjectionNotSupportedCheck())
      .withoutSemantic()
      .verifyNoIssues();
  }
  ```